api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java

```diff
@@ -680,7 +702,57 @@ public Long getCountOfPatients(String query, boolean includeVoided) {
     private List<Patient> findPatients(String query, boolean includeVoided) {
 		return findPatients(query, includeVoided, null, null);
 	}
-
+	
+	private List<Patient> findPatients(String query, List<PatientIdentifierType> identifierTypes, boolean matchExactly, Integer start, Integer length) {
+		String tmpQuery = query;
+		Integer tmpStart = start;
+		
+		if (tmpStart == null) {
+			tmpStart = 0;
+		}
+		Integer maxLength = HibernatePersonDAO.getMaximumSearchResults();
+		Integer tmpLength = length;
+		if (tmpLength == null || tmpLength > maxLength) {
+			tmpLength = maxLength;
+		}
+		tmpQuery = LuceneQuery.escapeQuery(tmpQuery);
+		
+		List<Patient> patients = new LinkedList<>();
+		
+		String minChars = Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_MIN_SEARCH_CHARACTERS);
+		
+		if (minChars == null || !StringUtils.isNumeric(minChars)) {
+			minChars = "" + OpenmrsConstants.GLOBAL_PROPERTY_DEFAULT_MIN_SEARCH_CHARACTERS;
+		}
+		if (tmpQuery.length() < Integer.valueOf(minChars)) {
+			return patients;
+		}
+		if(matchExactly) {
+			Context.getAdministrationService().setGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_IDENTIFIER_SEARCH_MATCH_MODE, OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_SEARCH_MATCH_EXACT);
```

Comment:
Do not use global property for a single invocation, because this setting will be set globally for all concurrent and subsequent invocations.

Pass the matchExactly to getPatientIdentifierLuceneQuery to override the GP setting for a single invocation.

Reply:
It was not my idea to use global properties but that is what was happening in the getPatientIdentifierLuceneQuery() method before I changed it. I adapted the code but did not delete the if statements checking for the global properties to not change existing code